lib/src/presentation/view/auth/login/login_view.dart

```diff
@@ -136,28 +136,50 @@ class _LoginViewState extends State<LoginView> {
                                       fontSize: 16.0,
                                     );
                                   } else {
-                                    Navigator.pushNamed(context,
-                                        RouterConstants.otpVerificationRoute,
-                                        arguments: phoneController.text);
+                                    
                                   }
                                 }
                               },
                               listenWhen: (previose, current) =>
                                   previose.isSendingOtp != current.isSendingOtp,
                               builder: (context, state) {
                                 if (state.isSendingOtp) {
-                                  return const  Center(
+                                  return const Center(
                                       child: CircularProgressIndicator());
                                 } else {
                                   return PrimaryButton(
                                     text: AppStrings.loginButtonLabel,
-                                    onPressed: () {
-                                      context.read<AuthBloc>().add(
-                                            AuthEvent.sendOtp(
-                                              phoneNumber: phoneController.text,
-                                            ),
-                                          );
-                                     
+                                    onPressed: () async {
+                                      if (formKey.currentState!.validate()) {
+                                        context.read<AuthBloc>().add(
+                                              AuthEvent.sendOtp(
+                                                phoneNumber:
+                                                    phoneController.text,
+                                              ),
+                                            );
+                                        final result =
+                                            await Navigator.pushNamed(
+                                          context,
+                                          RouterConstants.otpVerificationRoute,
```

Comment:
Why navigating to otp verification page just after validation ?

The navigation should be actioned only after the otp sending process status is success. Ad proper state management for the navigation after sending otp. 

Cross check the flow of state management with BloC